### PR TITLE
adding support for win/cygwin installations

### PR DESCRIPTION
**Pull request by jpkh**
adding support for cygwin installations for waf uploader

### CODE REVIEW

File: Tools/ardupilotwaf/px_uploader.py [hunk 1]
```diff
@@ -667,6 +668,9 @@ def main():
                         # OS X, don't open Windows and Linux ports
                         if "COM" not in port and "ACM" not in port:
                             up = uploader(port, args.baud_bootloader, baud_flightstack, args.baud_bootloader_flash)
+                    elif "cygwin" in _platform:
```
Comment (magicrub) [inline Tools/ardupilotwaf/px_uploader.py]: since it matches linux, why not make it:
`if "linux" in _platform or "cygwin" in _platform
`

**Comment by magicrub** [comment]
Good news is the port works and there's no python crashes anymore.

Bad news is I get into a boot loop. It reboots into the boot loader (which is good) which I know because the USB disconnects but then it seems like my bootloader moves on to the app and then it just reboots again.. the upload never begins.

I'm using a pixhawk1 connected over USB using /dev/ttyS13

Just in case, I tried setting --baud-flightstack 57600 and 115200

**Comment by jpkh** [comment]
@magicrub yes it goes on infinite loop if you dont have correct modules installen on python.

Please install pyserial in cygwin terminal: pip install pyserial

We are still missing that from cygwin doc and it will be added shortly 

Make sure that /dev/ has new ttySx when you connect your fliightboard. As example COM5 in win will be ttyS4 in cygwin

**Comment by jpkh** [comment]
No it does not match Linux, only works similar. Yes is was thinking to put it on same line as you did but wanted to keep it separate for future expansions.
For example we might want to move /dev/ttySx from default ports to cygwin only.

**Comment by magicrub** [comment]
confirmed working, merging. Thanks @jpkh !

**Comment by rmackay9** [comment]
@magicrub, @jpkh, let's remember to get the commit message right.

**Pull request merged**